irmaconfig.go

```diff
@@ -136,16 +134,6 @@ func (conf *Configuration) ParseFolder() (err error) {
 		if err != nil {
 			return err
 		}
-		if err = common.IterateSubfolders(conf.Path, func(dir string, _ os.FileInfo) error {
```

Comment:
If I remember correctly, the bug was that we vendored corrupted scheme assets in an app update. Then we removed the corrupted folder in an app update and we had to add code to remove all corrupted schemes from the assets.

Relaxing `ParseFolder` is not that trivial, because what should we do if a scheme becomes corrupted and we have credentials of that scheme in our storage. Then we cannot delete or disable a scheme, because then we would have credentials in our storage without valid configuration.

We could add a feature that `DisabledSchemeManagers` are ignored when doing a `loadCredentialStorage`, such that the app can always start. If we then make a similar mistake, then we can either supply the correct scheme via the assets or use the `RemoveScheme` function to remove the corrupt scheme.

I'm not sure how realistic this issue is to make dedicated support for this. It can be quite a hassle to make this. There are all kinds of corner cases, like what to do when the `pbdf` scheme is corrupt and it's also not vendored in the assets. Then the `irmaclient` could start in theory, but in irmamobile it will be impossible to enter your PIN.

Reply:
Yes, I agree with your final paragraph.

The commit that added this code is surrounded by commits introducing the requestor schemes, and if I recall correctly the bug then had something to do with those as opposed to ordinary issuer schemes. But if some problem causes something broken to be written to the schemes folder, then we can't know in advance what kind of scheme that will be. That means that we can't predict at this time the exact situation then, or how to recover from it. So I supppose the only reasonable behaviour then is to crash, which is what the code would do now.